lib/logzio-nodejs.js

```diff
@@ -265,33 +276,31 @@ class LogzioLogger {
         };
 
         if (typeof this.timeout !== 'undefined') {
-            options.timeout = this.timeout;
+            axios.defaults.timeout = this.timeout;
         }
 
         return Promise.resolve()
             .then(() => {
                 if (this.compress) {
-                    options.headers['content-encoding'] = 'gzip';
                     return zlibPromised(body);
                 }
                 return body;
             })
             .then((finalBody) => {
-                options.body = finalBody;
-                this._tryToSend(options, bulk);
+                this._tryToSend(finalBody, bulk);
             });
     }
 
-    _tryToSend(options, bulk) {
+    _tryToSend(body, bulk) {
         this._debug(`Sending bulk of ${bulk.msgs.length} logs`)
-        return request.post(options)
+        return axios.post(this.url, body)
             .then(() => {
                 this._debug(`Bulk #${bulk.id} - sent successfully`);
                 this.callback();
             })
             .catch((err) => {
                 // In rare cases server is busy
-                const errorCode = err.cause && err.cause.code;
+                const errorCode = err.code;
                 if (UNAVAILABLE_CODES.includes(errorCode)) {
```

Comment:
have u been able to generate one of the UNAVAILABLE_CODES?

Reply:
Yes
Based on the test `recovers after server fails one time => Msgs are only sent once`
I'm receiving `ECONNABORTED`
<img width="355" alt="Screen Shot 2022-07-12 at 10 02 47" src="https://user-images.githubusercontent.com/11467909/178429386-bbed890f-f6f7-4143-9c00-1d6240a3fa8f.png">